src/cargo_loading_service.cpp

```diff
@@ -12,150 +12,178 @@
+  // キャンセルになるまで設備連携要求を投げ続ける
+  while (!timer_->is_canceled()) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 1000 /* ms */, "request is running");
+  }
 
-  facility_info_.id = request->value;
+  // サービスのresult更新
+  response->state = service_result_;
 
-  {
-    std::lock_guard<std::mutex> lock(mutex_cargo_loading_state_);
-    facility_info_.approval = false;
-  }
+  // 各変数の初期化
+  service_result_ = ExecuteInParkingTask::Response::NONE;
+  facility_id_ = "";
+  infra_approval_ = false;
+}
 
-  // Publish command
-  while(cmd_pub) {
-    {
-      std::lock_guard<std::mutex> lock(mutex_parking_state_);
-      aw_state = aw_state_;
-    }
-    if (aw_state == InParkingStatus::AW_EMERGENCY) {
-      command.state = ERROR;
-    } else if (aw_state == InParkingStatus::AW_OUT_OF_PLACE) {
-      command.state = InfrastructureCommand::SEND_ZERO;
-    } else if (aw_state == InParkingStatus::AW_NONE) {
-      response_state = ExecuteInParkingTaskResponse::FAIL;
-      break;
-    } else {
-      command.state = REQUESTING;
-    }
-    InfrastructureCommandArray command_array;
-    auto stamp = this->get_clock()->now();
-    command.stamp = stamp;
-    command_array.commands.push_back(command);
-    command_array.stamp = stamp;
-    pub_cargo_loading_state_->publish(command_array);
-    rclcpp::sleep_for(command_pub_sleep_time_);
-    {
-      std::lock_guard<std::mutex> lock(mutex_cargo_loading_state_);
-      if (facility_info_.approval) {
-        cmd_pub = false;
-      }
-    }
-  }
+void CargoLoadingService::publishCommand(const uint8_t state)
+{
+  InfrastructureCommandArray command_array;
+  auto stamp = this->get_clock()->now();
+  command_array.stamp = stamp;
 
-  // Initialize class variables
-  facility_info_.id = "";
-  {
-    std::lock_guard<std::mutex> lock(mutex_cargo_loading_state_);
-    facility_info_.approval = false;
-  }
+  InfrastructureCommand command;
+  command.stamp = stamp;
+  command.type = COMMAND_TYPE;
+  command.id = facility_id_;
+  command.state = state;
 
-  // Response
-  response->state = response_state;
+  command_array.commands.push_back(command);
+  pub_commands_->publish(command_array);
 }
 
-void CargoLoadingService::onParkingState(
-  const in_parking_msgs::msg::InParkingStatus::ConstSharedPtr msg)
+void CargoLoadingService::onTimer()
 {
-  {
-    std::lock_guard<std::mutex> lock(mutex_parking_state_);
-    aw_state_ = msg->aw_state;
+  RCLCPP_INFO(this->get_logger(), "timer start");
+  // 設備連携が完了していない
+  if (!infra_approval_) {
+    // aw_stateで条件分岐
+    switch (aw_state_) {
+      // AWがEmergencyの場合はERRORを発出し続ける
+      case InParkingStatus::AW_EMERGENCY:
+        publishCommand(static_cast<std::underlying_type<CommandState>::type>(CommandState::ERROR));
+        RCLCPP_ERROR(this->get_logger(), "AW emergency");
+        break;
+      // AWが停留所外などではSEND_ZEROを発出し、設備連携結果はFAILで返す
+      case InParkingStatus::AW_OUT_OF_PARKING:
+      case InParkingStatus::AW_UNAVAILABLE:
```

Comment:
```suggestion
      case InParkingStatus::AW_UNAVAILABLE:
      case InParkingStatus::NONE:
```
https://github.com/eve-autonomy/cargo_loading_service/pull/2#discussion_r1114165828
のスレッドでのNONEの移動先はこちらで良いと思います。
この場合、設備に終了通知0b00を送り即時終了します。